src/plugins/intel_gpu/tests/fusions/fully_connected_fusion_test.cpp

```diff
@@ -529,5 +534,41 @@ INSTANTIATE_TEST_SUITE_P(fusings_gpu, fc_fp16_eltwise_prod, ::testing::ValuesIn(
     fully_connected_test_params{ CASE_FC_FP16_3D_2, 2, 3, "fully_connected_gpu_bfyx_ref" },
 }));
 
+class fc_fp16_convert_eltwise_fp32_fusing: public FullyConnectedFusingTestOneDNN {};
+TEST_P(fc_fp16_convert_eltwise_fp32_fusing, basic) {
+    auto p = GetParam();
+
+    // not fused: FullyConnected(FP16) - Reorder(FP32) - Eltwise(FP32)
+    //     fused: FullyConnected(FP32) - Eltwise(FP32)
+    create_topologies(
+        input_layout("input", get_input_layout(p)),
+        data("weights", get_mem(get_weights_layout(p))),
+        data("data", get_mem(layout{p.out_shape, data_types::f32, p.input_format})),
+        fully_connected("fc_prim", input_info("input"), "weights"),
+        reorder("convert", input_info("fc_prim"), p.default_format, data_types::f32),
+        eltwise("eltwise", { input_info("convert"), input_info("data") }, eltwise_mode::sum, cldnn::data_types::f32),
+        reorder("reorder_bfyx", input_info("eltwise"), p.default_format, data_types::f32)
+    );
+
+    tolerance = 1.f;
+
+    // Check if the output data type of FC is f32.
+    this->additional_test = [](network& not_fused, network& fused) {
+        for (auto& pi : fused.get_primitives_info()) {
+            if (pi.type_id == "fully_connected") {
+                ASSERT_EQ(pi.output_layout.data_type, data_types::f32);
+                break;
+            }
+        }
+    };
+
+    execute(p);
```

Comment:
What about checking fc output data type here(after execute), instead of passing it as `additional_test`?

Reply:
It's not possible because network instances are local variable of execute function.